### PR TITLE
add candidate register action serialization code

### PR DESCRIPTION
**Pull request by yutongp**

### CODE REVIEW

File: action/candidate_update.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			return err
+		}
+		cu.rewardAddress = rewardAddr
+	}
+	return nil
+}
+
+// IntrinsicGas returns the intrinsic gas of a CandidateUpdate
+func (cu *CandidateUpdate) IntrinsicGas() (uint64, error) {
+	return 0, nil
```
Comment (CoderZhi) [inline action/candidate_update.go]: is this a TODO?

File: action/candidate_update.go [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+}
+
+// Cost returns the total cost of a CandidateUpdate
+func (cu *CandidateUpdate) Cost() (*big.Int, error) {
+	return big.NewInt(0), nil
```
Comment (CoderZhi) [inline action/candidate_update.go]: same question here.

File: action/candidate_update.go [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+// Serialize returns a raw byte stream of the CandidateUpdate struct
+func (cu *CandidateUpdate) Serialize() []byte {
+	return byteutil.Must(proto.Marshal(cu.Proto()))
+}
+
+// Proto converts to protobuf CandidateUpdate Action
+func (cu *CandidateUpdate) Proto() *iotextypes.CandidateBasicInfo {
+	act := &iotextypes.CandidateBasicInfo{}
+
+	if len(cu.name) > 0 {
```
Comment (CoderZhi) [inline action/candidate_update.go]: redundant checking?

File: action/candidate_update.go (hunk 1 above)
Comment (yutongp) [reply action/candidate_update.go]: Referenced from other actions without payload. intrinsicgas is just set to 0

File: action/candidate_update.go (hunk 2 above)
Comment (yutongp) [reply action/candidate_update.go]: same

File: action/candidate_register.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	CandidateRegisterPayloadGas = uint64(100)
+	// CandidateRegisterBaseIntrinsicGas represents the base intrinsic gas for CandidateRegister
+	CandidateRegisterBaseIntrinsicGas = uint64(10000)
+)
+
+// CandidateRegister is the action to register a candidate
+type CandidateRegister struct {
+	AbstractAction
+
+	name            string
```
Comment (dustinxie) [inline action/candidate_register.go]: suggest to use CandName here
then we can use it to directly access the delegate map

File: action/candidate_register.go [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+// NewCandidateRegister creates a CandidateRegister instance
+func NewCandidateRegister(
+	nonce uint64,
+	name, operatorAddrStr, rewardAddrStr, ownerAddrStr, amountStr string,
+	duration uint32,
+	autoStake bool,
+	payload []byte,
+	gasLimit uint64,
+	gasPrice *big.Int,
+) (*CandidateRegister, error) {
```
Comment (dustinxie) [inline action/candidate_register.go]: validate name is 12-byte in length

File: action/candidate_update.go [hunk 6]
```diff
@@ -0,0 +1,129 @@
...
+	"github.com/iotexproject/iotex-core/pkg/util/byteutil"
+	"github.com/iotexproject/iotex-core/pkg/version"
+	"github.com/iotexproject/iotex-proto/golang/iotextypes"
+)
+
+// CandidateUpdate is the action to register a candidate
+type CandidateUpdate struct {
+	AbstractAction
+
+	name            string
```
Comment (dustinxie) [inline action/candidate_update.go]: same as above

**Pull request merged**